Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -149,14 +149,28 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
         // Propagate to User Action scope
         userActionScope = userActionScope?.scope(byPropagating: command, context: context, writer: writer)
 
-        // Send "application start" action if this is the very first view tracked in the app
         let hasSentNoViewUpdatesYet = version == 0
         if isInitialView, hasSentNoViewUpdatesYet {
-            sendApplicationStartAction(context: context, writer: writer)
+            needsViewUpdate = true
         }
 
         // Apply side effects
         switch command {
+        // Application Launch
+        case is RUMApplicationStartCommand:
+            sendApplicationStartAction(context: context, writer: writer)
+            if !isInitialView || viewPath != RUMOffViewEventsHandlingRule.Constants.applicationLaunchViewURL {
+                DD.telemetry.error(
+                    id: "application-start-error",
+                    message: "An RUMApplicationStartCommand got sent to something other than the ApplicationLaunch view.",
+                    kind: nil,
+                    stack: nil
+                )
```

Comment:
/nit You can use other signature available
```suggestion
                DD.telemetry.error("An RUMApplicationStartCommand got sent to something other than the ApplicationLaunch view.")
```
The `id` is inferred using `file:line:message`.

Reply:
Maybe we can add the command type as part of the message, I'm afraid `something` is not super actionable :)